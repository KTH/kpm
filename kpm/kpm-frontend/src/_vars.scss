$mblBreakpoint: "screen and (max-width: 980px)";
$prod: "https://app.kth.se/style/static/kth-style/img/kth-style";
$ref: "https://app-r.referens.sys.kth.se/style/static/kth-style/img/kth-style";

@function kthStyleUrl($path) {
  @return url($prod + $path), url($ref + $path);
}
