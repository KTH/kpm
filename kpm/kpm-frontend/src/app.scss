@use "stylesVars";
@use "stylesReset";
@use "stylesCanvas";
@use "vars" as *;
@use "@kth/style/scss/components/kpm";
@use "components/entrances.scss" as e;
@use "components/menu-item";

#kpm-6cf53 {
  pointer-events: none;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 9999;

  display: flex;
  flex-direction: column;
  align-items: center;

  font-family: "Figtree", sans-serif;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;

  @media print {
    display: none;
  }
}

#kpm-6cf53 .kth-kpm__container {
  width: 100%;
  pointer-events: all;
  gap: 0;
  overflow: hidden;
  // Overwrite KTH Style: minimum margin is `0rem` instead of `1rem`
  padding-inline: clamp(0rem, 3.33333vw, 2rem);
}

// Global styling
#kpm-6cf53 .kpm-modal {
  pointer-events: all;
  max-width: 1200px;
}

#kpm-6cf53 .kpm-modal-backdrop {
  pointer-events: all;
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: linear-gradient(rgba(0, 0, 0, 0.25), rgba(0, 0, 0, 0.45));
}

#kpm-6cf53 .kpm-row {
  display: flex;
  flex-flow: row wrap;
  width: 100%;
}

#kpm-6cf53 .kpm-col {
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  flex-grow: 1;

  .kpm-col-header {
    font-size: 1.17em;
    font-weight: bold;
    margin-bottom: var(--kpmMargin);
  }
}
@media #{$mblBreakpoint} {
  #kpm-6cf53 .kpm-col {
    margin-top: 1em;
    margin-bottom: 1em;
    flex-basis: 100%;
  }
}

#kpm-6cf53 .kpm-modal button.kpm-button,
#kpm-6cf53 .kpm-modal a.kpm-button {
  display: flex;
  padding: calc(var(--kpmPad)) calc(3 * var(--kpmPad));
  border-radius: var(--kpmButtonRadius);
  color: var(--kpmButtonText);
  background-color: var(--kpmButtonBg);
  border: 2px solid var(--kpmButtonFg);
  text-decoration: none;

  &:hover {
    color: var(--kpmButtonHoverText);
    background-color: var(--kpmButtonHoverBg);
    border-color: var(--kpmButtonHoverFg);
    text-decoration: none;
    cursor: pointer;
  }

  &.kpm-button-primary {
    --kpmButtonBg: var(--kpmPaneFg);
    --kpmButtonFg: var(--kpmPaneBg);
    --kpmButtonText: var(--kpmPaneBg);
  }

  &.kpm-button-link {
    --kpmButtonBg: var(--kpmPaneBg);
    --kpmButtonHoverBg: var(--kpmPaneBg);
    --kpmButtonFg: var(--kpmPaneBg);
    --kpmButtonHoverFg: var(--kpmPaneBg);
    --kpmButtonText: var(--kpmPaneFg);
    --kpmButtonHoverText: var(--kpmPaneText);
    padding: calc(var(--kpmPad) / 2) 0;
    margin: calc(var(--kpmPad) / 2) 0;
    border: 1px solid var(--kpmButtonBg);
    border-radius: 0;
    &:hover {
      border-bottom-color: var(--kpmButtonHoverText);
    }
  }
}

// "Login" link when user is logged out
#kpm-6cf53 .kpm-login {
  margin-inline-start: auto;
  margin-inline-end: 0.25rem;
}

// Typography in menu panes
#kpm-6cf53 .kpm-modal-content {
  padding: calc(2 * var(--kpmPad));

  h1 {
    margin-top: 0;
  }

  ul {
    margin: 0;
    padding: 0;
  }

  li {
    list-style: none;
    margin-bottom: var(--kpmMargin);
  }

  a.icon {
    color: var(--kpmLink);
    text-decoration: none;
    line-height: 1em;

    &:hover {
      color: var(--kpmLinkHover);
      text-decoration: underline;
    }
  }

  .kpm-muted-text {
    padding-top: 0.1em;
    color: var(--kpmPaneTextMuted);
    font-size: 0.9em;
  }
}

// This is used to detect css-breakpoints from TS-code
// See function isOnMobileBp()
#kpm-6cf53:before {
  content: "desktop";
  visibility: hidden;
  position: absolute;
  width: 0;
  height: 0;
}

@media #{$mblBreakpoint} {
  #kpm-6cf53:before {
    content: "mobile";
  }
}

@media #{$desktopLargeBreakpoint} {
  #kpm-6cf53:before {
    content: "desktop-large";
  }
}
