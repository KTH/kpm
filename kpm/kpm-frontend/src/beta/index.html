<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>KPM Beta Activation Page</title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
    />
    <link
      rel="shortcut icon"
      href="//www.kth.se/polopoly_fs/1.871948.1550155901!/favicon.ico"
    />
    <script src="https://app.kth.se/kpm/kpm.js" async></script>
    <!-- <link
      rel="stylesheet"
      href="https://app.kth.se/style/static/kth-style/css/kth-bootstrap.css?v=3.0.0-1670499721"
    /> -->
    <link rel="stylesheet" href="index.scss" />
    <script type="module" src="index.tsx"></script>
  </head>
  <body>
    <div class="kth-a11y-nav">
      <a href="#mainContent">Go to main content</a>
    </div>
    <div class="kth-kpm"></div>

    <div id="kpm-activation">Loading...</div>
  </body>
</html>
