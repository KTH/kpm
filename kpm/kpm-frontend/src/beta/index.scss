@use "../stylesReset";
@use "@kth/style/scss/components/a11y-nav";
