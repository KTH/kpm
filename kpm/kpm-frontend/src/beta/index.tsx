import * as React from "react";
import { createRoot } from "react-dom/client";
import { App } from "./App";

const appEl = document.getElementById("kpm-activation-6cf53");
const root = createRoot(appEl!);
root.render(<App />);
