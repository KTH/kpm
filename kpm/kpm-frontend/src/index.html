<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>KPM</title>
    <link rel="stylesheet" href="styles.scss" />
    <script type="module" src="app.tsx"></script>
  </head>
  <body>
    <div id="app">Loading...</div>
  </body>
</html>