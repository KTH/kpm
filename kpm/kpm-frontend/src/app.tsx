import * as React from "react";
import { createHashRouter, RouterProvider } from "react-router-dom";
import { getRoutes } from "./routes";
import { Menu } from "./Menu";
import { ErrorBoundary } from "./error";

import "./app.scss";
import { TSessionUser } from "kpm-backend-interface";

const IS_DEV = process.env.NODE_ENV !== "production";

type TSettings = {
  lang?: string;
};

declare global {
  interface Window {
    __kpmCurrentUser__: TSessionUser | undefined;
    __kpmSettings__: TSettings;
  }
}

function createRouter() {
  return createHashRouter([
    {
      path: "/",
      element: <Menu />,
      errorElement: <Menu />,
      children: getRoutes().map((route) => ({
        path: route.path,
        element: route.element,
      })),
    },
  ]);
}

export function App() {
  return (
    <ErrorBoundary>
      <RouterProvider router={createRouter()} />
    </ErrorBoundary>
  );
}
