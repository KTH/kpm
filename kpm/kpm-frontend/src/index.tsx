import * as React from "react";
import * as ReactDOM from "react-dom/client";
import { App } from "./app";

// Only mount menu in outermost frame
if (window.frameElement === null) {
  const root = ReactDOM.createRoot(document.getElementById("kpm-6cf53")!);
  root.render(<App />);
}
