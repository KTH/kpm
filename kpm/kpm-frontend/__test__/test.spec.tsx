import * as React from "react";
import { describe, expect, test } from "@jest/globals";
import renderer from "react-test-renderer";
import { App } from "../src/app";

describe("<App />", () => {
  test("Can be rendered", () => {
    const component = renderer.create(
      <App />,
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
