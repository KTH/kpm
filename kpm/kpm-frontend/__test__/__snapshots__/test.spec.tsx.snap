// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Can be rendered 1`] = `
<nav
  className="kpm-menu"
>
  <ul>
    <li
      className="kpm-mobile-menu kpm-mobile"
    >
      <a
        onClick={[Function]}
      >
        <img
          alt=""
          className="kpm-profile-image"
          src="https://www-r.referens.sys.kth.se/files/thumbnail/jhsware"
        />
        <span
          className="kpm-profile-display-name"
        >
          Dr.LocalDev Server
        </span>
      </a>
    </li>
    <li
      className="kpm-profile-item kpm-desktop"
    >
      <a
        className=""
        href="#/profile"
        onClick={[Function]}
      >
        <img
          alt=""
          className="kpm-profile-image"
          src="https://www-r.referens.sys.kth.se/files/thumbnail/jhsware"
        />
        <span
          className="kpm-profile-display-name"
        >
          Dr.LocalDev Server
        </span>
      </a>
    </li>
    <li
      className="kpm-schedule-item"
    >
      <a
        href="https://www.kth.se/social/home/calendar/"
      >
        Schedule
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/studies"
        onClick={[Function]}
      >
        Studies
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/teaching"
        onClick={[Function]}
      >
        Teaching
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/programme"
        onClick={[Function]}
      >
        Programme
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/groups"
        onClick={[Function]}
      >
        Groups
      </a>
    </li>
    <li>
      <a
        className=""
        href="#/services"
        onClick={[Function]}
      >
        Services
      </a>
    </li>
    <div
      className="kpm-menu-shotcuts"
    >
      <li
        className="kpm-try"
      >
        <a
          href="https://app.kth.se/kpm/"
          title="Give us feedback or disable the beta"
        >
          <span>
            â€½
          </span>
        </a>
      </li>
      <li>
        <a
          href="https://webmail.kth.se/"
        >
          <svg
            height="1em"
            version="1.1"
            viewBox="0 -100 1000 900"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              KTH Webmail
            </title>
            <path
              d="M 30,54 Q -2,36 2,14 4,0 28,0 h 846 q 38,0 20,32 -8,14 -24,22 -14,6 -192,102 -178,96 -182,98 -16,10 -46,10 -28,0 -46,-10 -4,-2 -182,-98 Q 44,60 30,54 Z m 850,100 q 20,-10 20,10 v 368 q 0,16 -17,32 -17,16 -33,16 H 50 Q 34,580 17,564 0,548 0,532 V 164 q 0,-20 20,-10 l 384,200 q 18,10 46,10 28,0 46,-10 z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="https://www.kth.se/social/home/subscriptions/"
        >
          <svg
            height="1em"
            version="1"
            viewBox="0 -100 1000 900"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              News feed
            </title>
            <desc>
              News from your courses, groups and programmes
            </desc>
            <path
              d="M290 460h350l6-2h4v92q0 40-29 70t-71 30H300L150 800V650h-50q-40 0-70-30T0 550V250q0-42 30-71t70-29h190zm610-560q42 0 71 29t29 71v300q0 40-29 70t-71 30h-50v150L700 400H350V0q0-42 30-71t70-29z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          className="icon kpm-has-new-notices"
          href="https://www.kth.se/social/notifications/notice_list/"
        >
          <svg
            height="1em"
            version="1"
            viewBox="0 -100 1000 900"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Direct notifications
            </title>
            <path
              d="M792 202q58 138 67 258t-39 140q-28 12-61-3t-65-40q-32-25-99-41t-149-8q-28 4-42 19t-6 37q22 56 46 108 4 10 24 22t24 20q14 34-22 46-50 22-102 40-30 10-54-42-32-76-58-132-6-12-34-17t-46-31l-38 14q-34 12-74-12t-54-60q-17-32-5-79t43-61q126-52 213-108t124-103q37-47 59-92t25-78q3-33 15-59t36-36q48-20 130 70t142 228zm-28 300q8-4 10-38t-11-98q-13-64-41-128-28-66-67-123t-67-84Q560 4 552 8t-10 42q-2 38 10 105t40 133q28 66 68 119t68 76q28 23 36 19z"
              fill="currentColor"
            />
          </svg>
          <span
            className="kpm-notice-new"
          >
            3
          </span>
          <span
            className="kpm-notice-new-trail"
          >
            new
          </span>
        </a>
      </li>
    </div>
    <li
      className="kpm-profile-item kpm-mobile"
    >
      <a
        className=""
        href="#/profile"
        onClick={[Function]}
      >
        <img
          alt=""
          className="kpm-profile-image"
          src="https://www-r.referens.sys.kth.se/files/thumbnail/jhsware"
        />
        Profile
      </a>
    </li>
    <li
      className="kpm-mobile-logout kpm-mobile"
    >
      <a
        href="/kpm/auth/logout"
      >
        Logout
      </a>
    </li>
  </ul>
  <div
    className="kpm-loading-indicator"
  />
</nav>
`;
